src/module/actor/SR5Actor.ts

```diff
@@ -871,9 +871,19 @@ export class SR5Actor extends Actor<ShadowrunActorData, SR5Item> {
                 if (!roll) return;
                 let hits = roll.total;
                 const data = this.data.data as CharacterData;
-                let current = data.track[track].value;
-
-                current = Math.max(current - hits, 0);
+                let current: number;
+                switch (track) {
+                    case 'stun':
+                        current = this.getStunTrack().max - data.track[track].value;
+                        break;
+                    case 'physical':
+                        current = data.track[track].value;
+                        break;
+                }
+                console.log(`Current value of ${track} is ${current}`);
+                
+                current = Math.min(current + hits, data.track[track].max);
+                console.log(`Modified current value of ${track} is ${current}`);
```

Comment:
Just noticed this seems like an odd change. Let me take another look at this (and at the very least remove the `console.log()` lines.

Reply:
Don't worry. I've been rather busy during this week but I'll have the review done before sunday. :)